src/main/scala/codacy/detekt/DocGenerator.scala

```diff
@@ -124,38 +122,31 @@ object DocGenerator {
   }
 
   private def generateRules: List[Rule] = {
-    val config = new YamlConfig((Map(("autoCorrect", false), ("failFast", false))).asJava, null)
-
-    _root_.codacy.helpers.ResourceHelper
-      .getResourceContent("META-INF/services/io.gitlab.arturbosch.detekt.api.RuleSetProvider")
-      .get //This is just a script, is better get the error
-      .view
-      .flatMap { clazz =>
-        Class
-          .forName(clazz)
-          .getDeclaredConstructor()
-          .newInstance()
-          .asInstanceOf[RuleSetProvider]
-          .instance(config)
-          .getRules
-          .asScala
-          .flatMap {
-            case r: MultiRule =>
-              r.asInstanceOf[MultiRule].getRules.asScala
-            case r: Rule =>
-              Seq(r.asInstanceOf[Rule])
-          }
-      }
-      .to(List)
+    val config = new YamlConfig(Map(("autoCorrect", false), ("failFast", false)).asJava, null)
+
+    val result = for {
+      providerClass <- Providers.classes
+      res <- providerClass
+        .newInstance()
```

Comment:
Use `.getDeclaredConstructor().newInstance()` instead of `.newInstance()`

Reply:
Simplified to not use reflection.